Automattic/woocommerce-payments
# Remove URL params from woopay_source_url

#### Changes proposed in this Pull Request
`woopay_source_url` that is used to track WooPay marketing consents, currently store the complete URL. This can break checkout when a store appends URL params and the URL is very long. 

This PR removes the URL parameters in the URL to reduce its length.









#### Testing instructions




To replicate URLs with params

1. Add a product to the cart and go to the checkout page
2. Edit the URL to include query params eg: 
http://wcpay.test/checkout/?param1=value&param2=value2
3. Check the Save My Info checkbox in WooPay
4. Complete the checkout
5. Go to your Stripe Connect account and notice the complete URL in the payment metafields

To replicate checkout error with long URLs.
1. In step 2 of the above, we need to exceed the URL length by over 500 characters. Add the following random query params to the URL:

```
?Fxxr6=JlFi1d9E&if2gR=BzPmFtrW&UAv4r=t2YirOYA&xlaDF=eFHedAd2&KIv5u=C5gar5ba&UrxJw=eYG3ngmC&iz1GS=5xxXfFYy&PPOML=2rPa1EIU&1oubG=Vc7T4QmX&Tzd69=h09pQoSY&2gnFa=gjj1U7d6&Ebkh8=lsoMolRO&Yctbt=Q0WPGbTg&zONK0=i7rBBFGU&zgxVf=ijvKZKgX&cd6Kf=XI25RWDC&l1TAG=fzGy1OAm&UML2n=rXWxRIBz&cYLen=QgrnnAC6&42uIT=SYatxDz0&jF62x=N4eFyt6p&pXFom=O65ytGND&Yk4um=n6VbHavk&kS4pW=m3ERrF1u&R67Wf=MyJcEBGC&GOpNB=4BY96NuH&Lz8y0=l5qKaxcR&bJEol=rxkZSNsW&LJu9i=W5dYCleA&4LQIx=2kuoUqap&jndNn=E8Gh6l7n&1st36=ugrtcPWd&T7mXA=XsOnkCMi&DynVm=3DEwpsXi
```

2. Continue from Step 3.
3. Notice that the order cannot be placed.

Test the fix
1. Repeat the above steps
4. Notice that the order can be placed and the query params are now removed from the URL stored in the `platform_checkout_source_url` metafield.


-------------------


**Post merge**
